### ISSUE TITLE
Loopy Dir doesn't refresh directory

### ISSUE THREAD

**Issue by BelowSubway** [AshMartian/ComfyUI-DirGir#9]
Hi there,

thanks for your custom node. I was looking for a node that could give me the latest file in a directory. Which your node did, I just had to set the loop_index on a fixed 0. There I just one little issue I have right now: Loopy Dir doesn't refresh itself, so it doesn't notice if a file gets added after it was loaded the first time.

Anyway to add a refresh to the node?

Thanks for your work.

**Comment by DeepSeaCatz** [comment]
Yeah, it seems to always use the loop_index: **0** as fixed number when you load comfy for the first time. Every time, I need to drag that value to a random number before the node works again. It's a bit annoying, I wonder what alternative there is to fix this.

**Comment by DeepSeaCatz** [comment]
@BelowSubway I've found a workaround. Convert the loop_index as input and feed it to any type of seed node. If you're using this multiple times, use a Seed Everywhere node to assign the seed to all of these. Somehow it still gives a random file, even if it's the same seed.
![image](https://github.com/user-attachments/assets/31711102-304e-427a-a48a-970cbccdcca5)

Oh and I've also replaced the GIR Dir Picker with a simple string since it wasn't letting me select a Directory and would always reset the value if copied and paste.

**Comment by AshMartian** [comment]
Will take a look into addressing this!

**Comment by AshMartian** [comment]
Fixed a few issues today:

1. Inputs into `loop_index` are respected
2. A new toggle `pause_loop` to not auto-increment
3. Removed the matched files cache mechanism, now each run the folder is re-scanned for files. 

Sorry it took so long!

**Comment by DeepSeaCatz** [comment]
Woohoo thanks! Love these nodes, so this helps a lot. Thanks for fixing them!
